docs/source/auth/index.rst

```diff
@@ -0,0 +1,613 @@
+:py:func:`AuthServiceClient<mlflow.server.auth.client.AuthServiceClient>` to manage users and permissions.
+To instantiate ``AuthServiceClient``, it is recommended that you use ``mlflow.server.get_app_client``.
+
+.. code-block:: python
+    :caption: Example
+
+    import os
+    from mlflow import MlflowClient
+    from mlflow.server import get_app_client
+
+    tracking_uri = "http://localhost:5000/"
+
+    auth_client = get_app_client("basic-auth", tracking_uri=tracking_uri)
+    auth_client.create_user(username="user1", password="pw1")
+    auth_client.create_user(username="user2", password="pw2")
+
+    os.environ["MLFLOW_TRACKING_USERNAME"] = "user1"
+    os.environ["MLFLOW_TRACKING_PASSWORD"] = "pw1"
+
+    client = MlflowClient(tracking_uri=tracking_uri)
+    experiment_id = client.create_experiment(name="experiment")
+
+    auth_client.create_experiment_permission(
+        experiment_id=experiment_id, username="user2", permission="MANAGE"
+    )
+
+
+Authenticating to MLflow
+========================
+
+Using MLflow UI
+---------------
+
+When a user first visits the MLflow UI on a browser, they will be prompted to login. 
+There is no limit to how many login attempts can be made.
+
+Currently, MLflow UI does not display any information of the current user.
+Once a user is logged in, the only way to log out is to close the browser.
+
+    .. image:: ../_static/images/auth_ui.png
+
+Using REST API
+--------------
+
+A user can authenticate using the HTTP ``Authorization`` request header.
+See https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication for more information.
+
+In Python, you can use the ``requests`` library:
+
+.. code-block:: python
+
+    import requests
+
+    response = requests.get(
+        "https://mlflow_tracking.uri/",
+        auth=("username", "password"),
+    )
+
+Using MLflow Client
+-------------------
+
+MLflow provides two environment variables for authentication: ``MLFLOW_TRACKING_USERNAME`` and ``MLFLOW_TRACKING_PASSWORD``.
+To use basic authentication, you must set both environment variables.
+
+.. code-block:: python
+
+    import os
+    import mlflow
+
+    os.environ["MLFLOW_TRACKING_USERNAME"] = "username"
+    os.environ["MLFLOW_TRACKING_PASSWORD"] = "password"
+
+    mlflow.set_tracking_uri("https://mlflow_tracking.uri/")
+    with mlflow.start_run():
+        ...
+
+
+Creating New User
+=================
+
+Using MLflow UI
+---------------
+
+MLflow UI provides a simple page for creating new users at ``<tracking_uri>/signup``.
+
+    .. image:: ../_static/images/auth_signup.png
+
+Using REST API
+--------------
+
+Alternatively, you can send ``POST`` requests to the Tracking Server endpoint ``2.0/users/create``.
+
+In Python, you can use the ``requests`` library:
+
+.. code-block:: python
+
+    import requests
+
+    response = requests.post(
+        "https://mlflow_tracking.uri/api/2.0/mlflow/users/create",
```

Comment:
```suggestion
        "https://<mlflow_tracking_uri>/api/2.0/mlflow/users/create",
```

Can we add `<>` here to clarify this is a placeholder that needs to be replaced?

Reply:
sure!